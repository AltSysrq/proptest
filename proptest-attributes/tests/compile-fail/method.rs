use proptest::{strategy::Strategy, string::string_regex};
use proptest_attributes::proptest;

#[derive(Debug, Clone, PartialEq)]
struct FunctionArn(String);

fn gen_function_arn() -> impl Strategy<Value = FunctionArn> {
    let expr = "arn:aws:lambda:us-east-1:[0-9]{12}:function:custom-runtime";
    let arn = string_regex(expr).unwrap();
    arn.prop_map(FunctionArn)
}

struct SomeStruct;

impl SomeStruct {
    #[proptest(gen_function_arn())]
    fn function_arn(&self) {
        let mut map = HashMap::new();
        map.insert("arn", arn.clone());
        prop_assert_eq!(map.get("arn"), Some(&arn));
    }
}

fn main() {}
