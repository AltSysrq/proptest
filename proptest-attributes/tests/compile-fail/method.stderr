error: The `#[proptest]` macro cannot be applied to a method.
 --> $DIR/method.rs:7:21
  |
7 |     fn function_arn(&self) {}
  |                     ^
